src/adaptations/device-layer/NetworkTelemetryManager.cpp

```diff
@@ -109,6 +112,39 @@ WEAVE_ERROR NetworkTelemetryManager::Init(void)
     return err;
 }
 
+void NetworkTelemetryManager::OnPlatformEvent(const WeaveDeviceEvent* event)
+{
+#if WEAVE_DEVICE_CONFIG_ENABLE_WIFI_TELEMETRY
+    if (event->Type == DeviceEventType::kWiFiConnectivityChange)
+    {
+        if (ConnectivityMgr().IsWiFiStationProvisioned() && ConnectivityMgr().IsWiFiStationEnabled())
+        {
+            mWiFiTelemetry.Enable();
```

Comment:
This has the effect of forcing telemetry on whenever WiFi connectivity changes.  Is that the correct thing to do in every case?  It effectively means the pubic Disable() function is useless, because it wont stay disabled for long.

Reply:
Is there a use case for disabling telemetry once it has been enabled ?

Reply:
Fixed